KIPs/kip-dreaft_sdk_common_architecture.md

```diff
@@ -0,0 +1,1623 @@
+| createWithAccountKeyRoleBased(address: String, roleBasedPubKeys: List&#60;String[]&#62;, options: List&#60;WeightedMultiSigOptions&#62;): Account | Creates an Account instance which has AccountKeyRoleBased as an accountKey. It throws an exception if the address string, public key strings or options are invalid. |
+| getRLPEncodingAccountKey(): String | Return RLP-encoded string of AccountKey in Account instance. |
+
+### Wallet Layer Class Diagram
+
+The `Wallet` layer allows the user to sign a message or a transaction through the Caver with a [Klaytn account].
+
+![0811Wallet](https://user-images.githubusercontent.com/32922423/89861019-c8b95300-dbdf-11ea-8aa6-30015a33100d.png)
+
+In the Wallet layer, an abstract class called `AbstractKeyring` is defined, and `SingleKeyring`, `MultipleKeyring` and `RoleBasedKeyring` are implemented by extending AbstractKeyring. AbstractKeyring defines abstract methods that Keyring classes must implement.
+
+`SingleKeyring` is a Keyring class that uses "only one private key", and `MultipleKeyring` is a Keyring class that uses "multiple private keys". `RoleBasedKeyring` is a Keyring class that uses "different private key(s) for each role", and the `keys` member variable is a 2D array in which keys to be used for each role are defined inside the array.
+
+Each Keyring class uses the `PrivateKey` class, which has one private key as a member variable.
+
+`KeyringFactory` provides static methods for creating `SingleKeyring`, `MultipleKeyring` or `RoleBasedKeyring`.
+
+`MessageSigned` is a structure that contains the result of signing a message and is used as a result of the `signMessage` function.
+
+`SignatureData` is a structure that stores a signature.
+
+`Keystore` is a class that contains encrypted keyring. The internally defined value differs depending on the keystore version, see [KIP-3](./kip-3.md).
+
+`KeyringContainer` is an "in-memory wallet" class that manages Keyring instances. Manage the Keyring instance using the address as the key value.
+
+#### PrivateKey
+
+`PrivateKey` is a class that contains a private key string. The private key to be used for each role in Keyring is defined as this `PrivateKey` instance.
+
+##### Variable description
+
+| Variable | Description |
+| ----------- | ----------- |
+| privateKey: String | The private key string. |
+
+##### Method description
+
+| Method | Description |
+| ----------- | ----------- |
+| sign(txHash: String, chainId: String): SignatureData | Signs the transaction. |
+| sign(txHash: String, chainId: int): SignatureData | Signs the transaction. |
+| signMessage(messageHash: String): SignatureData | Signs the message. |
+| getPublicKey(compressed: Boolean): String | Returns public key string. |
```

Comment:
```suggestion
| getPublicKey(compressed: Boolean): String | Returns the compressed public key string if compressed is true. It returns the uncompressed public key string otherwise. |
```